lms/services/assignment.py

```diff
@@ -0,0 +1,95 @@
+from functools import lru_cache
+
+from lms.models import ModuleItemConfiguration
+
+
+class AssignmentService:
+    """A service for getting and setting assignments (ModuleItemConfiguration's)."""
+
+    def __init__(self, db, tool_consumer_instance_guid, resource_link_id):
+        self._db = db
+        self._tool_consumer_instance_guid = tool_consumer_instance_guid
+        self._resource_link_id = resource_link_id
+
+    def get_document_url(self, tool_consumer_instance_guid=None, resource_link_id=None):
```

Comment:
I'm leaving these as `get_document_url()` and `set_document_url()` for now as changing that is going to raise questions that can be dealt with in a separate PR